packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/fpsrange/FpsRangeFeature.java

```diff
@@ -43,8 +47,9 @@ public FpsRangeFeature(@NonNull CameraProperties cameraProperties) {
       if (ranges != null) {
         for (Range<Integer> range : ranges) {
           int upper = range.getUpper();
-
-          if (upper >= 10) {
+          // When in photo mode, the upper bound is 30 fps or the aspect ratio/resolution will be
+          // changed by the camera session.
+          if (upper >= 10 && (upper < 60 || captureMode != CaptureMode.photo)) {
```

Comment:
Try as I might, I can't understand what's happening here. Maybe someone more familiar with the code would.